server/middleware/populateCurrentUser.ts

```diff
@@ -26,16 +26,16 @@ export default function populateCurrentUser(userService: UserService): RequestHa
     try {
       // Populate the currentUser details in the session if there is a token present and no user details
       if (res.locals.user?.token) {
-        const { token, username, authSource } = res.locals.user
+        const { username, authSource } = res.locals.user
         if (!req.session?.currentUser) {
           logger.info(`populateCurrentUser - populating ${authSource} user in session`)
           const cvlUser = new CvlUserDetails()
 
           if (authSource === 'nomis') {
             // Assemble user information from Nomis via prison API
             const [prisonUser, prisonUserCaseload] = await Promise.all([
-              userService.getPrisonUser(token),
-              userService.getPrisonUserCaseloads(token),
+              userService.getPrisonUser(username),
+              userService.getPrisonUserCaseloads(username),
             ])
```

Comment:
This was passing through the logged-in user's own token.

Reply:
The decision of whether or not to pass user token or system token is delegated to the API layer where it makes sense in my opinion